OpenRA.Mods.Common/Traits/Crates/RevealMapCrateAction.cs

```diff
@@ -31,18 +31,12 @@ public RevealMapCrateAction(Actor self, RevealMapCrateActionInfo info)
 			this.info = info;
 		}
 
-		bool ShouldReveal(Player collectingPlayer)
-		{
-			if (info.IncludeAllies)
-				return collectingPlayer.World.LocalPlayer != null &&
-					collectingPlayer.Stances[collectingPlayer.World.LocalPlayer] == Stance.Ally;
-
-			return collectingPlayer == collectingPlayer.World.LocalPlayer;
-		}
-
 		public override void Activate(Actor collector)
 		{
-			if (ShouldReveal(collector.Owner))
+			if (info.IncludeAllies)
+				foreach (var ally in collector.World.Players.Where(collector.Owner.IsAlliedWith))
```

Comment:
This could also be done without the LINQ query, by checking IsAlliedWith with an if inside foreach. Which one is better?